analytics/tests/test_views.py

```diff
@@ -714,6 +714,29 @@ def test_activate_or_deactivate_realm(self) -> None:
             m.assert_called_once_with(lear_realm)
             self.assert_in_success_response(["Realm reactivation email sent to admins of lear"], result)
 
+    def test_change_subdomain(self) -> None:
+        cordelia = self.example_user('cordelia')
+        lear_realm = get_realm('lear')
+        self.login_user(cordelia)
+        current_subdomain = lear_realm.string_id
+
+        result = self.client_post("/activity/support", {"realm_id": f"{lear_realm.id}",
+                                                        "new_subdomain": "new_name"})
+        self.assertEqual(result.status_code, 302)
+        self.assertEqual(result["Location"], "/login/")
+        self.login('iago')
+
+        result = self.client_post("/activity/support", {"realm_id": f"{lear_realm.id}", "new_subdomain": "new_name"})
+        self.assert_in_success_response([f"Subdomain changed from {current_subdomain} to new_name"], result)
+        lear_realm = get_realm('new_name')
+        self.assertEqual(lear_realm.string_id, "new_name")
+
+        result = self.client_post("/activity/support", {"realm_id": f"{lear_realm.id}", "new_subdomain": "new_name"})
+        self.assert_in_success_response(["Realm with subdomain new_name already exists!"], result)
```

Comment:
The error message for this case should be more explicit. It should say that the new subdomain should not be the same as the old subdomain.

Reply:
please correct me if I am wrong... when the new_subdomain and old_subdomain are same, then the error message should be "The new subdomain should not be same as old subdomain" ... and when the new_subdomain is already in use, then the error message should "Realm with {new_subdomain} already exists!" ... is that correct??? I have made the changes in the views.py and the test based on this assumption.